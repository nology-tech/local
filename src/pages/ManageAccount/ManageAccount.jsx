import "./ManageAccount.scss";
import Layout from "../../components/Layout/Layout";

import Button from "../../components/Button/Button";

const ManageAccount = () => {
  return (
    <Layout isManageAccount={true} isWithMenu={true}>
      <div className="manage-account">
        <h1 className="manage-account__heading">Account</h1>
        <div className="manage-account__button-container">
          <div className="manage-account__buttons-updateAccount">
            <Button
              buttonName="account button--signout"
              buttonText="Sign Out"
            />
            <Button
              buttonName="account button--update-details"
              buttonText="Update details"
            />
          </div>
          <div className="manage-account__buttons-deleteAccount">
            <Button
              buttonName="account button--deleteAccount"
              buttonText="Delete Account"
            />
          </div>
        </div>
      </div>
    </Layout>
  );
};
export default ManageAccount;
