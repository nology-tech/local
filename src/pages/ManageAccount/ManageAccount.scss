@use "../../styles/variables" as *;

.manage-account {
  height: 100%;
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: center;
  background-color: $white;
}
.manage-account__heading {
  margin-left: 1.5rem;
  font-family: $header-font-family;
  font-weight: $large-font-weight;
  font-size: $title-font-size;
}

.manage-account__button-container {
  height: 70%;
  width: min-content;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}
