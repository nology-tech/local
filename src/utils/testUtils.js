import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
