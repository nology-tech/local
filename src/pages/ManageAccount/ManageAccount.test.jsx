import { customRender } from "../../utils/testUtils";
import ManageAccount from "./ManageAccount";

it("should render the text on the display", () => {
  const { container } = customRender(<ManageAccount />);

  expect(container).toMatchSnapshot();
});
