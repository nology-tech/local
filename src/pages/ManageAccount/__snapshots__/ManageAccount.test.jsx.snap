// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout layout__manage-account"
  >
    <div
      class="header header__manage-account"
    >
      <img
        alt=""
        src="BackButton.svg"
      />
      <div
        class="header__container"
      >
        <div
          class="header__logo-title"
        >
          <img
            alt=""
            src="AppLogo.svg"
          />
          <h1
            class="header__title-local"
          >
            LOCAL
          </h1>
        </div>
      </div>
      <div
        class="menu-icon__container"
        role="button"
      >
        <svg
          alt="Menu-icon"
          class="menu-icon__icon"
          fill="none"
          height="30"
          viewBox="0 0 44 30"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 30H44V25H0V30ZM0 17.5H44V12.5H0V17.5ZM0 0V5H44V0H22H0Z"
            fill="black"
          />
        </svg>
      </div>
    </div>
    <div
      class="layout__menu menu-closed"
    >
      <div
        class="menu"
      >
        <div
          class="menu__header"
        >
          <img
            alt="App Logo"
            class="menu__app-logo"
            src="logo.svg"
          />
          <h1
            class="menu__app-title"
          >
            LOCAL
          </h1>
          <img
            alt="Close Button"
            class="menu__close"
            src="close-btn.svg"
          />
        </div>
        <div
          class="menu__home"
        >
          <a
            class="menu__link"
            href="/home"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="home.svg"
              />
              <h2
                class="menu__option"
              >
                Home
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__favourites"
        >
          <a
            class="menu__link"
            href="/favourites"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="favourites.svg"
              />
              <h2
                class="menu__option"
              >
                Favourites
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__info"
        >
          <a
            class="menu__link"
            href="/account"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Account Logo"
                class="menu__logo"
                src="account.svg"
              />
              <h2
                class="menu__option"
              >
                Manage Account
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__about"
        >
          <a
            class="menu__link"
            href="/about"
          >
            <div
              class="menu__page"
            >
              <img
                alt="About Logo"
                class="menu__logo"
                src="about.svg"
              />
              <h2
                class="menu__option"
              >
                About
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__contact"
        >
          <a
            class="menu__link"
            href="/contact"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Contact Logo"
                class="menu__logo"
                src="contact.svg"
              />
              <h2
                class="menu__option"
              >
                Contact
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="manage-account"
    >
      <h1
        class="manage-account__heading"
      >
        Account
      </h1>
      <div
        class="manage-account__button-container"
      >
        <div
          class="manage-account__buttons-updateAccount"
        >
          <button
            class="button button--account button--signout"
          >
            Sign Out
          </button>
          <button
            class="button button--account button--update-details"
          >
            Update details
          </button>
        </div>
        <div
          class="manage-account__buttons-deleteAccount"
        >
          <button
            class="button button--account button--deleteAccount"
          >
            Delete Account
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
